flytekit/core/type_engine.py

```diff
@@ -401,14 +403,22 @@ def get_transformer(cls, python_type: Type) -> TypeTransformer[T]:
             TODO lets make this deterministic by using an ordered dict
 
         """
+
         # Step 1
         if python_type in cls._REGISTRY:
             return cls._REGISTRY[python_type]
 
         # Step 2
         if hasattr(python_type, "__origin__"):
+
+            # Handling of annotated generics, eg:
+            # typing.Annotated[typing.List[int], 'foo']
+            if isinstance(python_type, typing._AnnotatedAlias):
```

Comment:
we should use the [get_origin](https://docs.python.org/3/library/typing.html#typing.get_origin) and check if the resulting value is `typing.Annotated`. For example:

```
>>> get_origin(Annotated[List[int], 'annotation_0', 'annotation_1']) is Annotated
True
>>> get_origin(List[int]) is Annotated
False
>>> get_origin(int) is Annotated
False
>>> get_origin(Annotated[int, 'a', 42]) is Annotated
True
>>> get_origin(Annotated[Dict[str, int], 'dict_annotation']) is Annotated
True
```

Reply:
is there a reason? is there a case where checking for `typing._AnnotatedAlias` would fail but this approach would catch it? or conversely any false positives that this approach would avoid? or is strictly style

Reply:
again, the idea is to bullet-proof this code. If you look at the [list of symbols exported](https://github.com/python/cpython/blob/3.9/Lib/typing.py#L32-L121) by the typing module, `_AnnotatedAlias` is not present.

Reply:
also makes sense